pool/database.go

```diff
@@ -0,0 +1,67 @@
+package pool
+
+import (
+	"net/http"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+// Database describes all of the functionality needed by a dcrpool database
+// implementation.
+type Database interface {
+	// Utils
+	httpBackup(w http.ResponseWriter) error
+	backup(fileName string) error
+	close() error
+
+	// Pool metadata
+	persistPoolMode(mode uint32) error
+	fetchCSRFSecret() ([]byte, error)
+	persistCSRFSecret(secret []byte) error
+	persistLastPaymentInfo(height uint32, paidOn int64) error
+	loadLastPaymentInfo() (uint32, int64, error)
+	persistLastPaymentCreatedOn(createdOn int64) error
+	loadLastPaymentCreatedOn() (int64, error)
+
+	// Account
+	FetchAccount(id string) (*Account, error)
```

Comment:
I suspect almost all the functions required by the interface here could be unexported. Will look into doing that if you decide against it in this PR.